heron/common/src/cpp/config/topology-config-helper.h

```diff
@@ -74,6 +75,22 @@ class TopologyConfigHelper {
 
   // Gets the per container ram requested by this topology
   static sp_int64 GetContainerRamRequested(const proto::api::Topology& _topology);
+
+  // Is the topology stateful
+  static bool IsTopologyStateful(const proto::api::Topology& _topology);
+
+  // Is the topology exactly once
+  static bool IsTopologyExactlyOnce(const proto::api::Topology& _topology);
```

Comment:
This will be required in stmgrs to see whether marker propagation needs to be done